client/src/index.js

```diff
@@ -27,8 +27,8 @@ import Auth from './modules/Auth';
 
 //remove tap delay, essential for MaterialUI to work properly
 injectTapEventPlugin();
-
-const store = createStore(reducer, applyMiddleware(ReduxPromise));
+const createStoreWithMiddleware = applyMiddleware(ReduxPromise)(createStore);
+const store = createStoreWithMiddleware(reducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
```

Comment:
This is how we are able to add many middlewares to your store while using some cool plugin action

See https://github.com/gaearon/redux-devtools for more information.